### DISCUSSION TITLE
How to reuse same charts multiple times in one dashboards?

### DISCUSSION THREAD

**Discussion by matejkratochvil** [apache/superset#18693] [Q&A / Help]
Hello, could someone help me to find out how to put same charts in one dashboard? It seems that it is not possible as once the chart is added to the dashboard, it turns gray in the charts tab and cannot be added anymore.
My usecase would be to have same chart with different filters applied, for example for displaying same metrics in different segments or for different products. 
The only way how to achieve this I came up with is to copy each single chart (k1+....+kn)-times where ki is the number of values of the variable i by which i filter the data. 
This seem kind of suboptimal given the capabilities of dashboard filters in superset.
Thanks a lot.

**Comment by raags** [comment]
I would like the same. From a functional point of view, it doesn't seem this should be a problem. It would also be good to see visual if the same chart is being used elsewhere in the dashboard, maybe via an icon that shows the number of times, it's used, as a hint, when editing the dashboard.

**Comment by crazyisjen** [comment]
i experienced too...cant add the same chart to dashboard

**Comment by fernando-freitas-alves** [comment]
up

**Comment by TheHassi** [comment]
I would also be happy about this feature.

**Comment by LeoDiep** [comment]
> Hello, could someone help me to find out how to put same charts in one dashboard? It seems that it is not possible as once the chart is added to the dashboard, it turns gray in the charts tab and cannot be added anymore. My usecase would be to have same chart with different filters applied, for example for displaying same metrics in different segments or for different products. The only way how to achieve this I came up with is to copy each single chart (k1+....+kn)-times where ki is the number of values of the variable i by which i filter the data. This seem kind of suboptimal given the capabilities of dashboard filters in superset. Thanks a lot.

View chart in explore >> change the title of the chart >> Save chart as... (DON'T overwrite) >> Chose the destination dashboard as the same dashboard. Done

**Reply by awoehrl** [reply]
I think this is the only solution right now, however I don't think it's feasible in many use cases. The charts become very much polluted with the same chart over and over again containing just slight variations in a filter.

**Reply by LeoDiep** [reply]
I use this when the new chart is totally different to the original one (different metrics, different type of charts,...)
For variations in filter, you might want to look at Jinja template:
https://www.youtube.com/watch?v=1QpG_m_ix3w

**Comment by Stefan781** [comment]
+1
Having to duplicate charts is not ideal. If you make a change to the chart, you have to repeat the process.

**Comment by goto-loop** [comment]
This would be a great feature. Duplicating the chart is not an elegant solution and not really feasible when you have many charts and several copies of each.

**Comment by bkampsnl** [comment]
Yes this would be a great feature!

**Comment by ofey404** [comment]
+1, longing for this feature.

**Comment by theblinkingusb** [comment]
There is a backend way to do this
On the /api/v1/dashboard/ route, or the public.dashboards table there is a position_json field which describes the layout of the dashboard. Inside of that object one must create a new row object and reference the existing chart key (it is probably easiest to start with a template dashboard). One can then add the new row object to the children within the GRID_ID prop.
I've renamed the chart/row keys on these sample objects - one may see hash-like property names when pulling the position_json field initially. 
[pre_position_json.json](https://github.com/apache/superset/files/15143757/pre_position_json.json)
[post_position_json.json](https://github.com/apache/superset/files/15143758/post_position_json.json)

Update the db field or send the request back to the api and the same chart will appear multiple times on that dashboard.

Another additional but likely unnecessary step is duplicating the chart object (again referencing the same slice) and pointing the new row reference to that chart id.

**Reply by XiangRongLin** [reply]
I tried something similiar with the export/import function and while it does work, there are problems with the filters.

I can't have have both charts prefiltered in different ways. are you able to also solve this?

**Comment by sandeepr43** [comment]
+1

**Comment by dlg0** [comment]
+1

**Comment by gpandolfino** [comment]
+1

**Reply by readyon-reza** [reply]
+1

**Comment by opvexe** [comment]
+1

**Comment by Tchinkatchuk** [comment]
Must have feature in terme of dashboard quality + efficiency

**Comment by toransahu** [comment]
+1

**Comment by shengbinxu** [comment]
+1

**Comment by David-Parro-P** [comment]
+1

**Comment by ondrej-sulc-payx** [comment]
+1

**Comment by dorian-monkey** [comment]
+1

**Comment by alfredho811** [comment]
+1 Please

**Comment by sfirke** [comment]
There is clearly a lot of demand for this feature!

Anyone from the community is welcome to take a stab at implementing this. If you have questions or want to talk through aspects of your design and implementation, you could comment here or come to a SIP Review or Townhall meeting (here's the Superset community calendar: https://calendar.google.com/calendar/u/0/embed?src=superset.committers@gmail.com).  

Or if you feel confident, feel free to just open up a pull request!

**Comment by nono-london** [comment]
currently testing version 6.00 and it has not been added... The general experience is great though..
